internal/discovery/namespace_filter.go

```diff
@@ -0,0 +1,126 @@
+package discovery
+
+import (
+	"errors"
+
+	"github.com/newrelic/nri-kubernetes/v3/internal/config"
+	"github.com/newrelic/nri-kubernetes/v3/internal/storer"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+)
+
+// NamespaceFilterer provides an interface to filter namespaces.
+type NamespaceFilterer interface {
+	IsAllowed(namespace string) bool
+}
+
+// NamespaceFilter is a struct holding pointers to the config and the namespace lister.
+type NamespaceFilter struct {
+	c      *config.Config
+	lister listersv1.NamespaceLister
+	stopCh chan<- struct{}
+	storer storer.Storer
```

Comment:
not sure if this is the storer interface that this filter needs. It looks a bit odd that returns a value while this needs a bool i think

Reply:
Would you suggest to implement a new cache?

Reply:
What you can do, to continue using the same storer is creating another interface:
```
 type NamespaceStorer interface {
	Set(key string, value bool) int64
	Get(key string, valuePtr *bool) (int64, error)
}
```
And then create a wrapper that basically does:
```
type NamespaceInMemoryStore struct {
	InMemoryStore
}

func (ns *NamespaceInMemroyStore) func Set(key string, value bool) int64 {
   ns.InMemoryStore.Set(key, value)
}
[...]
```
WDYT?

Reply:
Finally I implemented a new cache.